nengo/builder/connection.py

```diff
@@ -19,17 +20,16 @@
     'BuiltConnection', ['decoders', 'eval_points', 'transform', 'solver_info'])
 
 
-def build_linear_system(model, conn):
+def build_linear_system(model, conn, rng):
     encoders = model.params[conn.pre_obj].encoders
     gain = model.params[conn.pre_obj].gain
     bias = model.params[conn.pre_obj].bias
 
-    eval_points = conn.eval_points
-    if eval_points is None:
+    if conn.eval_points is None:
         eval_points = npext.array(
             model.params[conn.pre_obj].eval_points, min_dims=2)
     else:
-        eval_points = npext.array(eval_points, min_dims=2)
+        eval_points = gen_eval_points(conn.pre_obj, conn.eval_points, rng)
```

Comment:
If `conn.pre_obj` is not an ensemble, this will give a weird error when we try to get `n_eval_points` from it. Since we'll never need eval points if we don't have an ensemble, I'd put in something like `conn.pre_obj.n_eval_points if isinstance(conn.pre_obj, Ensemble) else 0'